---
int32[] error_id
int32 res